src/ParallelAlgorithms/LinearSolver/Tags.hpp

```diff
@@ -100,6 +100,24 @@ struct Residual : db::PrefixTag, db::SimpleTag {
   using tag = Tag;
 };
 
+/// Compute the residual \f$r=b - Ax\f$ from the `SourceTag` \f$b\f$ and the
+/// `db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo, FieldsTag>`
+/// \f$Ax\f$.
+template <typename FieldsTag, typename SourceTag>
+struct ResidualCompute : db::add_tag_prefix<Residual, FieldsTag>,
+                         db::ComputeTag {
+  using base = db::add_tag_prefix<Residual, FieldsTag>;
+  using type = db::item_type<base>;
+  using argument_tags =
+      tmpl::list<SourceTag, db::add_tag_prefix<OperatorAppliedTo, FieldsTag>>;
+  static type function(
```

Comment:
please make this a mutating compute tag; non-mutating compute tags will disappear in the near future